lib/cb/exceptions.rb

```diff
@@ -11,7 +11,12 @@
 module Cb
   class IncomingParamIsWrongTypeException < StandardError; end
   class ExpectedResponseFieldMissing < StandardError; end
-  class ApiResponseError < StandardError; end
+  class ApiResponseError < StandardError
+    attr_accessor :raw_response, :response, :code
+  end
   class ServiceUnavailableError < ApiResponseError; end
   class UnauthorizedError < ApiResponseError; end
+  class DocumentNotFoundError < ApiResponseError; end
+  class BadRequestError < ApiResponseError; end
+  class ServerError < ApiResponseError; end
```

Comment:
Style guide related question: The guide suggests using `ServerError = Class.new(ApiResponseError)` for class definitions with no body. Is that something we want to follow? Would that syntax work here?

https://github.com/cbdr/ruby-style-guide#single-line-classes

(Obviously this should not hold up this PR)